yamcs-client/src/main/java/org/yamcs/client/utils/WellKnownTypes.java

```diff
@@ -51,10 +48,23 @@ public static ListValue toListValue(Object[] array) {
         return listb.build();
     }
 
+    public static String toStringValue(byte[] array) {
+        char[] HEXCHARS = "0123456789abcdef".toCharArray();
+        char[] hexChars = new char[array.length * 2];
+        for (int j = 0; j < array.length; j++) {
+            int v = array[j] & 0xFF;
+            hexChars[j * 2] = HEXCHARS[v >>> 4];
+            hexChars[j * 2 + 1] = HEXCHARS[v & 0x0F];
+        }
+        return new String(hexChars);
+    }
+
     @SuppressWarnings("unchecked")
     public static Value toValue(Object value) {
         if (value == null) {
             return Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build();
+        } else if (value instanceof byte[]) {
```

Comment:
not sure why you convert a byte[] into a StringValue. We also have a Binary Value.

Reply:
Fabian pointed me in this direction. I do not seem to find a binary value that I can use.